### PR TITLE
Initial implementation

### PR DESCRIPTION
**Pull request by DoGab**
## Checklist

- [x] Keep pull requests small so they can be easily reviewed.
- [x] Update the documentation.
- [x] Update the ./CHANGELOG.md.
- [x] Link this PR to related issues.

### CODE REVIEW

File: class/defaults.yml [hunk 1]
```diff
@@ -1,3 +1,55 @@
 parameters:
   podsecuritypolicy:
     namespace: syn-podsecuritypolicy
+    policies:
+      syn-privileged:
```
Comment (srueg) [inline class/defaults.yml]:
```suggestion
      syn-privileged: # https://raw.githubusercontent.com/kubernetes/website/master/content/en/examples/policy/privileged-psp.yaml
```

File: class/defaults.yml [hunk 2]
```diff
@@ -1,3 +1,55 @@
...
+        hostPID: true
+        runAsUser:
+          rule: 'RunAsAny'
+        seLinux:
+          rule: 'RunAsAny'
+        supplementalGroups:
+          rule: 'RunAsAny'
+        fsGroup:
+          rule: 'RunAsAny'
+      syn-restricted:
```
Comment (srueg) [inline class/defaults.yml]:
```suggestion
      syn-restricted: # https://raw.githubusercontent.com/kubernetes/website/master/content/en/examples/policy/restricted-psp.yaml
```

File: lib/podsecuritypolicy.libsonnet [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+ * \file Helper to create PodSecurityPolicy objects.
+ */
+
+local com = import 'lib/commodore.libjsonnet';
+local kap = import 'lib/kapitan.libjsonnet';
+local kube = import 'lib/kube.libjsonnet';
+local inv = kap.inventory();
+local params = inv.parameters.podsecuritypolicy;
+
+local podSecurityPolicy(name) = kube._Object('policy/v1beta1', 'PodSecurityPolicy', name) {
```
Comment (srueg) [inline lib/podsecuritypolicy.libsonnet:20]: Please add the same comments as here for example: https://github.com/projectsyn/component-storageclass/blob/master/lib/storageclass.libsonnet#L12

**Review by srueg — CHANGES_REQUESTED** [review]

File: lib/podsecuritypolicy.libsonnet (hunk 3 above)
Comment (DoGab) [reply lib/podsecuritypolicy.libsonnet:20]: done

File: component/main.jsonnet [hunk 4]
```diff
@@ -1,10 +1,65 @@
...
-// Define outputs below
 {
+  [name]:
+    local policy = params.policies[name];
+    local labels = {
+      'app.kubernetes.io/managed-by': 'syn',
+    };
+    local cluster_role = kube.ClusterRole('psp:' + name) {
+      metadata+: {
+        labels: labels,
```
Comment (srueg) [inline component/main.jsonnet]:
```suggestion
        labels+: labels,
```

File: component/main.jsonnet [hunk 5]
```diff
@@ -1,10 +1,65 @@
...
+        metadata+: {
+          annotations+: annotations,
+        },
+        spec+: com.makeMergeable(policy.spec),
+      },
+      cluster_role,
+    ] + if std.objectHas(policy, 'bindToGroups') then [
+      kube.ClusterRoleBinding(name) {
+        metadata+: {
+          labels: labels,
```
Comment (srueg) [inline component/main.jsonnet]:
```suggestion
          labels+: labels,
```

**Review by srueg — CHANGES_REQUESTED** [review]

File: component/main.jsonnet (hunk 5 above)
Comment (DoGab) [reply component/main.jsonnet]: done

File: component/main.jsonnet (hunk 4 above)
Comment (DoGab) [reply component/main.jsonnet]: done

**Review by srueg — APPROVED** [review]

**Pull request merged**